### PR TITLE
Spaceship sprite

### PR DESCRIPTION
**Pull request by dasadboy**
implemented basic field
implemented spaceship class
- gets drawn to screen
- rotates and faces 180 degrees from cursor when left mouse button is held down
- moves in the direction it's facing when left mouse button is released but only on x-axis for now (movement in y-axis will be implemented in field)

### CODE REVIEW

File: src/ship.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#include "ship.h"
+
+Ship::Ship() {
+    this->radius = SHIP_CONSTS::SHIP_RADIUS;
+    this->health = SHIP_CONSTS::INITIAL_HEALTH;
+
+    if (!(this->texture).loadFromFile(SHIP_CONSTS::PATH_TO_TEXTURE, SHIP_CONSTS::TEXTURE_RECT))
```
Comment (aa6moham) [inline src/ship.cpp:14]: ideally not being able to load a texture should be a game ending condition so I would prefer if you move this out of the constructor and into a seperate init() function and check for ret code for better error handling.

File: src/game.cpp [hunk 2]
```diff
@@ -1 +1,71 @@
-#include "game.h"
\ No newline at end of file
+#include "game.h"
+
+#define PI 3.14159265f
+#define CCW90 90
```
Comment (aa6moham) [inline src/game.cpp]: why isnt this a float as well?

File: src/game.cpp [hunk 3]
```diff
@@ -1 +1,71 @@
...
+Game::Game()
+{
+    this->LMBHeldDown = false;
+    this->velocity = 0.f;
+    this->angle = 90.f;
+}
+
+void Game::init()
+{
+    this->window.create(sf::VideoMode(DISPLAY_CONSTS::WIDTH, DISPLAY_CONSTS::HEIGHT), "FeelsRocketMan");
```
Comment (aa6moham) [inline src/game.cpp]: FeelsRocketMan should be a game constant

File: src/game.cpp [hunk 4]
```diff
@@ -1 +1,71 @@
...
+        while (this->window.pollEvent(event))
+        {
+            if (event.type == sf::Event::Closed)
+            {
+                terminate();
+            }
+            else if (event.type == sf::Event::MouseButtonReleased && event.mouseButton.button == sf::Mouse::Left)
+            {
+                auto [shipx, shipy] = this->ship.getPosition();
+                auto [mousex, mousey] = sf::Mouse::getPosition(this->window);
```
Comment (aa6moham) [inline src/game.cpp:38]: isn't mousex and mousey already float, is this static_cast really necessary?

File: src/game.cpp [hunk 5]
```diff
@@ -1 +1,71 @@
...
+                this->velocity = SHIP_CONSTS::THRUST_L1;
+                this->angle = (std::atan2(shipy -  static_cast<float>(mousey), shipx - static_cast<float>(mousex)) * 180.f/(PI));
+                this->LMBHeldDown = false;
+            }
+            else if ((event.type == sf::Event::MouseButtonPressed && event.mouseButton.button == sf::Mouse::Left) || this->LMBHeldDown == true)
+            {
+                auto [shipx, shipy] = this->ship.getPosition();
+                auto [mousex, mousey] = sf::Mouse::getPosition(this->window);
+                this->ship.trackMouse(std::atan2(shipy -  static_cast<float>(mousey), shipx - static_cast<float>(mousex)) * 180.f/(PI) + CCW90);
```
Comment (aa6moham) [inline src/game.cpp]: seems like this conversion to degrees from radians and vice versa will be common operations that you'll be doing, why not define a macros for it?

File: src/ship.cpp [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+void Ship::trackMouse(float deg)
+{
+    shipSprite.setRotation(deg);
+}
+
+void Ship::move(float dx) 
+{
+    auto [currX, currY]  = this->shipSprite.getPosition();
+    float x = this->shipSprite.getPosition().x + dx;
```
Comment (aa6moham) [inline src/ship.cpp]: why aren't you using currX here?

File: includes/ship.h [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        unsigned int health;
+        sf::Texture texture;
+        sf::CircleShape shipSprite;
+        float vel;
+
+    public:
+
+        Ship();
+
+        bool collides(sf::Vector2f& pixelPos);
```
Comment (aa6moham) [inline includes/ship.h]: collidies is just a check so it should be a const function as well.

File: src/ship.cpp [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    this->shipSprite.setPosition(x + (x < 0) * DISPLAY_CONSTS::WIDTH - (x >= DISPLAY_CONSTS::WIDTH) * DISPLAY_CONSTS::WIDTH, currY);
```
Comment (aa6moham) [inline src/ship.cpp:41]: not sure what the x expression is for.

**Review by aa6moham — CHANGES_REQUESTED** [review]

File: src/game.cpp (hunk 4 above)
Comment (dasadboy) [reply src/game.cpp:38]: Mouse::getPosition returns ints.

File: src/ship.cpp (hunk 8 above)
Comment (dasadboy) [reply src/ship.cpp:41]: % doesn't work on floats so I have to use this funky expression to keep it within bounds.

**Review by aa6moham — APPROVED** [review]

**Pull request merged**